bindings/pyroot/pythonizations/python/ROOT/_pythonization/_roofit/_roodataset.py

```diff
@@ -117,30 +116,27 @@ def select_range_and_change_type(arr, dtype):
             if range_mask is not None:
                 arr = arr[range_mask]
             arr = arr if arr.dtype == dtype else np.array(arr, dtype=dtype)
-            return arr
-
-        for real in dataset.store().realStoreList():
-            vec = real.data()
-            arg = real.bufArg()
-            arr = select_range_and_change_type(data[arg.GetName()], np.float64)
-
-            vec.resize(len(arr))
-
-            beg = arr.ctypes.data_as(ctypes.POINTER(ctypes.c_double))
-            void_p = ctypes.cast(beg, ctypes.c_voidp).value + 8 * len(arr)
-            end = ctypes.cast(void_p, ctypes.POINTER(ctypes.c_double))
-            ROOT.std.copy(beg, end, vec.begin())
-
-        for cat in dataset.store().catStoreList():
-            vec = cat.data()
-            arg = cat.bufArg()
-            arr = select_range_and_change_type(data[arg.GetName()], np.int32)
-            vec.resize(len(arr))
-
-            beg = arr.ctypes.data_as(ctypes.POINTER(ctypes.c_int))
-            void_p = ctypes.cast(beg, ctypes.c_voidp).value + 4 * len(arr)
-            end = ctypes.cast(void_p, ctypes.POINTER(ctypes.c_int))
-            ROOT.std.copy(beg, end, vec.begin())
+            # Make sure that the array is contiguous to we can std::copy() it.
+            # In the implementation of ascontiguousarray(), no copy is done if
+            # the array is already contiguous, which is exactly what we want.
+            return np.ascontiguousarray(arr)
+
+        def copy_to_dataset(store_list, np_type, c_type, type_size_in_bytes):
+            for real in store_list:
+                vec = real.data()
+                arg = real.bufArg()
+                arr = select_range_and_change_type(data[arg.GetName()], np_type)
+
+                vec.resize(len(arr))
+
+                beg = arr.ctypes.data_as(ctypes.POINTER(c_type))
```

Comment:
Possibly add comment here 
```suggestion
                # The next part works because arr is guaranteed to be C-contiguous
                beg = arr.ctypes.data_as(ctypes.POINTER(c_type))
```